domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmResourceDefinition.java

```diff
@@ -48,6 +50,17 @@
  */
 public class SecurityRealmResourceDefinition extends SimpleResourceDefinition {
 
+    static {
+        //register the Openssl Provider, if possible
+        //not really sure if this is the best place for it
```

Comment:
I would suggest not including this within this commit.

We still need to rework it further but the Elytron subsystem contains resource(s) to locate and optionally register security providers - we can also make use of discovered providers without actually registering them globally.